runtime/compiler/trj9/runtime/J9Profiler.cpp

```diff
@@ -2415,10 +2426,15 @@ TR_PersistentProfileInfo::decRefCount(TR_PersistentProfileInfo *info)
    {
    VM_AtomicSupport::subtract((uintptr_t*) &(info->_refCount), 1);
    TR_ASSERT_FATAL(info->_refCount >= 0, "Decrement resulted in negative reference count");
-   if (info->_refCount == 0 && !TR::Options::getCmdLineOptions()->getOption(TR_DisableProfilingDataReclamation))
+   if (!TR::Options::getCmdLineOptions()->getOption(TR_DisableJProfilerThread))
+      {
+      if (info->_refCount == 0 && TR::Options::isAnyVerboseOptionSet(TR_VerboseProfiling, TR_VerboseReclamation))
+         TR_VerboseLog::writeLineLocked(TR_Vlog_RECLAMATION, "PersistentProfileInfo 0x%p queued for reclamation.", info);
```

Comment:
Don't you have to set the `_active` flag to false here?

Reply:
The `_active` flag is not related to whether or not the `TR_PersistentProfileInfo` is reclaimed. It controls whether the `TR_JProfilerThread` inspects its information for potential resets. Only the reference count is inspected when deciding to reclaim.